backend/src/modules/elimination-tournament-tree/core/domain/TournamentTree.ts

```diff
@@ -0,0 +1,114 @@
+import {TournamentTeam} from "./TournamentTeam";
+import {EntityIdGenerator} from "../../../../shared/core/application/EntityIdGenerator";
+import {FightingTeamsGroup} from "./FightingTeamsGroup";
+import {FightingTeamsGroupId} from "./FightingTeamsGroupId";
+import {toUnicode} from "punycode";
+
+export class TournamentTree {
+    readonly tournamentTeams: TournamentTeam[];
+
+    constructor(
+        props: { tournamentTeams: TournamentTeam[] }
+    ) {
+        this.tournamentTeams = props.tournamentTeams;
+    }
+}
+
+export function createTournamentTree(
+    props: {
+        tournamentTeams: TournamentTeam[],
+        entityIdGenerator: EntityIdGenerator,
+    }
+): FightingTeamsGroup[] {
+
+    const numberOfFightingTeamsGroups = fightingTeamsGroupsNumber(props.tournamentTeams.length);
+    const winnerTree = createWinnerTree(props.tournamentTeams, numberOfFightingTeamsGroups, props.entityIdGenerator);
+    return winnerTree;
+}
+
+function createWinnerTree(tournamentTeams: TournamentTeam[], fightingTeamsGroupsNumber: number, entityIdGenerator: EntityIdGenerator,): FightingTeamsGroup[] {
```

Comment:
Ogólnie zdaje mi się, że wszystkie algorytmy na drzewie są implementowane łatwiej w sposób rekurencyjny :) 
Ale... ja potrzebuję wytłumaczenia tego kodu :P Najlepiej na spotkaniu, bo nie wszystko rozumiem.

Reply:
https://github.com/nowakprojects/CodersCamp2020.Project.FullStack-Node-React.TableSoccerTournaments/issues/2

Pamiętaj o tym co podrzucałem wyżej, bo algorytm tego drzewa jest znany i już nie raz sotosowany - można się posiłkować tym jak to inni zrobili.
Chwalebne jest kombinować samemu i może to jest dobrze :) Ale przyda się Ania i lepsze zrozumienie kodu.

Reply:
W sumie teraz to widze, że to jest nawet bardziej Graf.

Reply:
Dzięki za podpowiedzi. Przeglądałem te materiały kilka razy - jeszcze przejrzę - ale nie do końca rozumiem.
Aktualnie po obejrzeniu struktur drzewek dla 4, 8 ,16 ,32 drużyn zauważyłem reguły matematyczne i jestem w stanie przelać to na papier, tfu... na kod. 
Ale właśnie widzę, że jest łatwiejszy (znany)
 sposób zapisu tylko nie mogę tego zrozumieć.

Reply:
@nowakprojects Czy z rekurencją chodzi o to Tobie by obiekt meczu zawierał w sobie obiekt potomnego meczu?
Może moge to tak rozbudować